hack-sound-server

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+
+import gi
+import json
+import os
+import signal
+import uuid
+gi.require_version('GLib', '2.0')  # noqa
+gi.require_version('Json', '1.0')  # noqa
+from gi.repository import Gio
+from gi.repository import GLib
+from gi.repository import Gst
+from gi.repository import Json
+
+
+def get_datadir():
+    return os.path.join(GLib.get_user_data_dir(), "hack-sound-server")
+
+def get_metadata_path():
+    return os.path.join(get_datadir(), "metadata.json")
+
+def get_sounds_dir():
+    return os.path.join(get_datadir(), "sounds")
+
+
+class HackSoundPlayer:
+    def __init__(self, metadata, sender):
+        self.metadata = metadata
+        self.sender = sender
+        self.pipeline = self._build_pipeline()
+
+    def play(self):
+        self.pipeline.set_state(Gst.State.PLAYING)
+
+    @property
+    def sound_location(self):
+        return os.path.join(get_sounds_dir(), self.metadata["sound-file"])
+
+    def _build_pipeline(self):
+        print(self.sound_location)
+        spipeline = "filesrc location=\"%s\" ! decodebin ! autoaudiosink" % self.sound_location
+        print(spipeline)
+        return Gst.parse_launch(spipeline)
+
+
+class HackSoundServer(Gio.Application):
+
+    _DBUS_NAME = "com.endlessm.HackSoundServer"
+    _DBUS_XML = """
+    <node>
+      <interface name='com.endlessm.HackSoundServer'>
+        <method name='PlaySound'>
+          <arg type='s' name='sound_event' direction='in'/>
+          <arg type='s' name='uuid' direction='out'/>
```

Comment:
What do you intend to do with the UUID? is it a key with which you can later replay the sound or stop the sound from looping, for example?

I notice the player is kept in the `self.players` dict when the sound is finished, but a new player is created if you subsequently play the same sound again. So we can get an unbounded number of players, which is not good.

Reply:
Yes, that's to stop the sound later.

Reply:
For that purpose, it would probably be more future-proof to return the object path of another exported DBus object with an interface that can stop the playback. That way, we could potentially also expose/change some of the metadata properties at runtime if it's necessary.
However it's OK if we want to keep this very simple for now.